### PR TITLE
Missing "Examples" headings and a few more actual examples

### PR DESCRIPTION
**Pull request by kshyatt**
Pretty minimal changes for the most part, some doctests were missing their headings! I added a few more examples to simple math functions where the docstring mentioned a behaviour that wasn't shown in the doctest. This passed `make docs` locally, be sure to edit out the `ci skip` whoever merges.

### CODE REVIEW

File: base/math.jl [hunk 1]
```diff
@@ -182,12 +186,27 @@ log(b::T, x::T) where {T<:Number} = log(x)/log(b)
...
 
 julia> log(4,2)
 0.5
+
+julia> log(-2, 3)
+ERROR: DomainError with log:
+-2.0 will only return a complex result if called with a complex argument. Try -2.0(Complex(x)).
+Stacktrace:
+ [1] throw_complex_domainerror(::Float64, ::Symbol) at ./math.jl:31
+ ...
```
[the remark this answers is not in the document]
Comment (fredrikekre) [reply base/math.jl]: Also, to make this doctest more resilient to line changes you can use a doctest-filter. You can use the same as here: https://github.com/JuliaLang/julia/blob/22590d529dceed93ae1dd8f32d569edba3be9f50/base/essentials.jl#L447

File: base/math.jl [hunk 2]
```diff
@@ -395,6 +418,13 @@ julia> log2(4)
 
 julia> log2(10)
 3.321928094887362
+
+julia> log2(-2)
+ERROR: DomainError with -2.0:
+NaN result for non-NaN input.
+Stacktrace:
+ [1] nan_dom_err at ./math.jl:325 [inlined]
+ ...
```
Comment (fredrikekre) [inline base/math.jl]: `[...]` and add a stacktrace filter.

File: base/math.jl [hunk 3]
```diff
@@ -412,6 +442,13 @@ julia> log10(100)
 
 julia> log10(2)
 0.3010299956639812
+
+julia> log10(-2)
+ERROR: DomainError with -2.0:
+NaN result for non-NaN input.
+Stacktrace:
+ [1] nan_dom_err at ./math.jl:325 [inlined]
+ ...
```
Comment (fredrikekre) [inline base/math.jl]: `[...]` and add a stacktrace filter.

File: base/math.jl [hunk 4]
```diff
@@ -429,6 +466,13 @@ julia> log1p(-0.5)
 
 julia> log1p(0)
 0.0
+
+julia> log1p(-2)
+ERROR: DomainError with log1p:
+-2.0 will only return a complex result if called with a complex argument. Try -2.0(Complex(x)).
+Stacktrace:
+ [1] throw_complex_domainerror(::Float64, ::Symbol) at ./math.jl:31
+ ...
```
Comment (fredrikekre) [inline base/math.jl]: `[...]` and add a stacktrace filter.

File: base/math.jl [hunk 5]
```diff
@@ -450,6 +494,22 @@ end
...
+```jldoctest
+julia> sqrt(big(81))
+9.0
+
+julia> sqrt(big(-81))
+ERROR: DomainError with -8.1e+01:
+NaN result for non-NaN input.
+Stacktrace:
+ [1] sqrt(::BigFloat) at ./mpfr.jl:501
+ ...
```
Comment (fredrikekre) [inline base/math.jl]: `[...]` and add a stacktrace filter.

File: base/dict.jl [hunk 6]
```diff
@@ -534,8 +536,9 @@ end
 """
     haskey(collection, key) -> Bool
 
-Determine whether a collection has a mapping for a given key.
+Determine whether a collection has a mapping for a given `key`.
```
[the remark this answers is not in the document]
Comment (StefanKarpinski) [reply base/dict.jl]: Seems like it works as either.

**Comment by kshyatt** [comment]
Are we happy with this? Good to sq&m?

**Comment by fredrikekre** [comment]
I get some doctest failure on this branch.

**Comment by fredrikekre** [comment]
#27109 on Travis 64-bit

**Pull request merged**